### PR TITLE
Remove the old sampler utilities

### PR DESCRIPTION
**Pull request by mattdangerw**
The code developed here all lives on in the new sampler API!

### CODE REVIEW

File: benchmarks/text_generation.py [hunk 1]
```diff
@@ -20,11 +20,6 @@
 from tensorflow import keras
```
Comment (chenmoneygithub) [inline benchmarks/text_generation.py:20]: We need to update this README with new metrics: https://github.com/keras-team/keras-nlp/blob/master/benchmarks/README.md

**Review by chenmoneygithub — CHANGES_REQUESTED** [review]
Thanks!

**Comment by jbischof** [comment]
Love the work thank you!

File: benchmarks/README.md [hunk 2]
```diff
@@ -11,16 +11,15 @@ from the root of the repository:
 python3 ./keras_nlp/benchmarks/text_generation.py
 ```
 
-On running this script on Google Colab (with Tesla T4 GPU, and TensorFlow 2.10.0),
+On running this script on Google Colab (with 3090 GPU, and TensorFlow 2.11.0),
```
Comment (chenmoneygithub) [inline benchmarks/README.md:14]: I should probably upgrade my PC 🛩️

**Review by chenmoneygithub — APPROVED** [review]

**Comment by abheesht17** [comment]
Minor comment, but this change will break this example (after release): https://keras.io/examples/nlp/neural_machine_translation_with_keras_nlp/. Considering this example is kinda popular, we should open a "contributions welcome" issue for this.

File: benchmarks/README.md [hunk 3]
```diff
@@ -11,16 +11,15 @@ from the root of the repository:
...
 |:---------------------:	|:--------------------:	|:-----------------------------:	|
-|     Greedy Search     	|        495.78        	|             293.77            	|
-|      Beam Search      	|        564.23        	|             615.17            	|
-|     Random Search     	|        446.55        	|             296.21            	|
-|      Top-k Search     	|        458.68        	|             302.66            	|
-|      Top-p Search     	|        468.63        	|             565.50             	|
+|     Greedy Search     	|        470.23        	|              61.79            	|
+|      Beam Search      	|        530.13        	|             189.61            	|
+|      Top-k Search     	|        374.05        	|              62.87            	|
+|      Top-p Search     	|        401.97        	|             260.31             	|
```
Comment (abheesht17) [inline benchmarks/README.md:22]: Damn, crazy speedup!

**Comment by mattdangerw** [comment]
> Minor comment, but this change will break this example (after release): https://keras.io/examples/nlp/neural_machine_translation_with_keras_nlp/. Considering this example is kinda popular, we should open a "contributions welcome" issue for this.

Yeah absolutely. Was going to file a contributor issue to do this as https://github.com/keras-team/keras-nlp/pull/896 lands. Doesn't make too much sense to do while we are still tweaking the sampler interface.

**Pull request merged**